discord/flags.py

```diff
@@ -140,6 +141,44 @@ def __iter__(self) -> Iterator[Tuple[str, bool]]:
             if isinstance(value, flag_value):
                 yield (name, self._has_flag(value.flag))
 
+    def __and__(self, other):
+        if isinstance(other, self.__class__):
+            return self.__class__._from_value(self.value & other.value)
+        elif isinstance(other, flag_value):
+            return self.__class__._from_value(self.value & other.flag)
+        else:
+            raise TypeError(f"'&' not supported between instances of {type(self)} and {type(other)}")
+
+    def __or__(self, other):
+        if isinstance(other, self.__class__):
+            return self.__class__._from_value(self.value | other.value)
+        elif isinstance(other, flag_value):
+            return self.__class__._from_value(self.value | other.flag)
+        else:
+            raise TypeError(f"'|' not supported between instances of {type(self)} and {type(other)}")
+
+    def __add__(self, other):
+        try:
+            return self | other
+        except TypeError:
+            raise TypeError(f"'+' not supported between instances of {type(self)} and {type(other)}")
+
+    def __sub__(self, other):
+        if isinstance(other, self.__class__):
+            return self.__class__._from_value(self.value & ~other.value)
+        elif isinstance(other, flag_value):
+            return self.__class__._from_value(self.value & ~other.flag)
+        else:
+            raise TypeError(f"'-' not supported between instances of {type(self)} and {type(other)}")
+
+    def __invert__(self):
+        return self.__class__._from_value(~self.value)
+
+    __rand__: Callable[[BaseFlags], bool] = __and__
+    __ror__: Callable[[BaseFlags], bool] = __or__
+    __radd__: Callable[[BaseFlags], bool] = __add__
+    __rsub__: Callable[[BaseFlags], bool] = __sub__
```

Comment:
I'm a little confused with these type annotations. The `__and__` and `__or__` functions support BaseFlags and flag_value for the `other` parameter

Reply:
Same confusion here